infra/label_evaluator_step_functions/lambdas/unified_receipt_evaluator.py

```
@@ -0,0 +1,975 @@
+
+    except Exception as e:
+        from receipt_agent.utils.llm_factory import AllProvidersFailedError
+        from receipt_agent.utils.ollama_rate_limit import (
+            OllamaRateLimitError,
+        )
+
+        if isinstance(e, (OllamaRateLimitError, AllProvidersFailedError)):
+            logger.error(
+                "Rate limit error, propagating for Step Function retry: %s",
+                e,
+            )
```

Ruff (0.14.11)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)